### PR TITLE
Improve Object.prototype.isPrototypeOf coverage

### PR DESCRIPTION
**Pull request by shvaikalesh**
JSC bug: [Object.prototype.isPrototypeOf() should check if the passed in value is a non-object first](https://bugs.webkit.org/show_bug.cgi?id=204971).

### CODE REVIEW

File: test/built-ins/Object/prototype/isPrototypeOf/builtin.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2009 the Sputnik authors. All rights reserved.
```
Comment (leobalter) [inline test/built-ins/Object/prototype/isPrototypeOf/builtin.js]:
```suggestion
// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
```

File: test/built-ins/Object/prototype/isPrototypeOf/context-is-in-prototype-chain-of-value.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2009 the Sputnik authors. All rights reserved.
```
Comment (leobalter) [inline test/built-ins/Object/prototype/isPrototypeOf/context-is-in-prototype-chain-of-value.js]:
```suggestion
// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
```

File: test/built-ins/Object/prototype/isPrototypeOf/length.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2009 the Sputnik authors. All rights reserved.
```
Comment (leobalter) [inline test/built-ins/Object/prototype/isPrototypeOf/length.js]:
```suggestion
// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
```

File: test/built-ins/Object/prototype/isPrototypeOf/undefined-context-and-object-value-throws.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2011 the Sputnik authors. All rights reserved.
```
Comment (leobalter) [inline test/built-ins/Object/prototype/isPrototypeOf/undefined-context-and-object-value-throws.js]:
```suggestion
// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
```

File: test/built-ins/Object/prototype/isPrototypeOf/null-context-and-primitive-value-returns-false.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  the this value is undefined or null.
+info: |
+  Object.prototype.isPrototypeOf ( V )
+
+  1. If Type(V) is not Object, return false.
+  2. Let O be ? ToObject(this value).
+features: [Symbol]
+---*/
+
+assert.sameValue(Object.prototype.isPrototypeOf.call(null, undefined), false);
```
Comment (leobalter) [inline test/built-ins/Object/prototype/isPrototypeOf/null-context-and-primitive-value-returns-false.js]: This is a very nice test, btw!

File: test/built-ins/Object/prototype/isPrototypeOf/null-context-and-object-value-throws.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2011 the Sputnik authors. All rights reserved.
```
Comment (leobalter) [inline test/built-ins/Object/prototype/isPrototypeOf/null-context-and-object-value-throws.js]: Just another nit pick: avoid referencing `this` as `context`. They are not fully the same thing spec wise.

Ideally the file names would be like `null-this-and-object-arg-throws.js`. See this file name also specify the arg is the object.

File: test/built-ins/Object/prototype/isPrototypeOf/builtin.js (hunk 1 above)
Comment (shvaikalesh) [reply test/built-ins/Object/prototype/isPrototypeOf/builtin.js]: Please see aec7c87: this isn't a new test, but rather merged and refactored [`S15.2.4.6_A6.js`](https://github.com/tc39/test262/blob/e316773ad4715bb6a10c67698fd61776510f3e56/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A6.js) and [`S15.2.4.6_A7.js`](https://github.com/tc39/test262/blob/e316773ad4715bb6a10c67698fd61776510f3e56/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A7.js), authored by Sputnik team.

File: test/built-ins/Object/prototype/isPrototypeOf/context-is-in-prototype-chain-of-value.js (hunk 2 above)
Comment (shvaikalesh) [reply test/built-ins/Object/prototype/isPrototypeOf/context-is-in-prototype-chain-of-value.js]: Please see b3b31b4: this isn't a new test, but rather cleaned up and renamed [`S15.2.4.6_A1.js`](https://github.com/tc39/test262/blob/52fd2522b1b1c22d488382be5513c78e327aa853/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A1.js), authored by Sputnik team.

File: test/built-ins/Object/prototype/isPrototypeOf/length.js (hunk 3 above)
Comment (shvaikalesh) [reply test/built-ins/Object/prototype/isPrototypeOf/length.js]: Please see 8ce3790: this test isn't new, but rather merged and refactored to use new harness [`S15.2.4.6_A10.js`](https://github.com/tc39/test262/blob/3aa02c9bbe2f9b1cdb200f356f7ab6ae3693e3ed/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A10.js), [`S15.2.4.6_A11.js`](https://github.com/tc39/test262/blob/3aa02c9bbe2f9b1cdb200f356f7ab6ae3693e3ed/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A11.js), [`S15.2.4.6_A8.js`](https://github.com/tc39/test262/blob/3aa02c9bbe2f9b1cdb200f356f7ab6ae3693e3ed/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A8.js), and [`S15.2.4.6_A9.js`](https://github.com/tc39/test262/blob/3aa02c9bbe2f9b1cdb200f356f7ab6ae3693e3ed/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A9.js), all authored by Sputnik team.

File: test/built-ins/Object/prototype/isPrototypeOf/undefined-context-and-object-value-throws.js (hunk 4 above)
Comment (shvaikalesh) [reply test/built-ins/Object/prototype/isPrototypeOf/undefined-context-and-object-value-throws.js]: Please see 86ba713: this test isn't new, but rather renamed [`S15.2.4.6_A12.js`](https://github.com/tc39/test262/blob/f505a9c9c32c4d2335aa53c2347524cc935ecd75/test/built-ins/Object/prototype/isPrototypeOf/S15.2.4.6_A12.js), authored by Sputnik team. Same goes for [`null-this-and-object-arg-throws.js`](https://github.com/tc39/test262/blob/86ba7134d51c89850dd76bfef723f586a7f07dd7/test/built-ins/Object/prototype/isPrototypeOf/null-this-and-object-arg-throws.js).

File: test/built-ins/Object/prototype/isPrototypeOf/null-context-and-object-value-throws.js (hunk 6 above)
Comment (shvaikalesh) [reply test/built-ins/Object/prototype/isPrototypeOf/null-context-and-object-value-throws.js]: Thank you, I've renamed `context` to `this` and `value` to `arg` & rebased.

**Pull request merged**